src/main/java/duke/Storage.java

```diff
@@ -0,0 +1,152 @@
+import java.util.Arrays;
+import java.util.List;
+import java.util.Scanner;
+
+/**
+ * Storage is a class to load and save tasks.
+ */
+public class Storage {
+    private Path relativeDirectoryPath;
+    private Path relativeFilePath;
+    private Path filepath;
+
+    /**
+     * Constructor for Storage.
+     * @param projectRoot filepath for storage.
+     */
+    public Storage(String projectRoot) {
+        this.relativeDirectoryPath = Paths.get(projectRoot, "data");
+        this.relativeFilePath = Paths.get(projectRoot, "data", "duke.txt");
+        this.filepath = Paths.get(projectRoot);
+    }
+
+    /**
+     * Returns a task that is parsed from string form.
+     * @param details details of the task.
+     * @param taskType type of task.
+     * @return task object.
+     */
+    public Task parseStringToTask(String details, String taskType) {
+        Task task = null;
+        switch (taskType) {
+        case "T": {
+            String[] taskDetails = details.split(" \\| ");
+            task = new ToDo(taskDetails[1]);
+            if (Integer.parseInt(taskDetails[0]) == 1) {
+                task.markAsDone();
+            }
+            break;
+        }
+        case "D":
+        case "E": {
+            String[] taskDetails = details.split(" \\| ");
+            task = new Deadline(taskDetails[1], taskDetails[2]);
+            if (Integer.parseInt(taskDetails[0]) == 1) {
+                task.markAsDone();
+            }
+            break;
+        }
+        default:
+            break;
+        }
+        return task;
+    }
+
+    /**
+     * Returns an ArrayList containing task objects that were previously saved.
+     * @return ArrayList containing task objects.
+     * @throws IOException
+     */
+    public ArrayList<Task> load() throws IOException {
+        if (!Files.exists(relativeDirectoryPath)) {
+            Files.createDirectory(relativeDirectoryPath);
+        }
+        if (!Files.exists(relativeFilePath)) {
+            Files.createFile(relativeFilePath);
+        }
+        ArrayList<Task> tasks = new ArrayList<>();
+        Scanner sc = new Scanner(new File(String.valueOf(relativeFilePath)));
+        ArrayList<String> stringTasks = new ArrayList<>();
+        while(sc.hasNextLine()) {
+            stringTasks.add(sc.nextLine());
+        }
+        for (String stringTask : stringTasks) {
+            String[] taskDetails = stringTask.split(" \\| ", 2);
+            Task task = null;
+            String taskType = taskDetails[0];
+            switch (taskType) {
+            case "T": {
+                task = parseStringToTask(taskDetails[1], "T");
+                break;
+            }
+            case "D": {
+                task = parseStringToTask(taskDetails[1], "D");
+                break;
+            }
+            case "E": {
+                task = parseStringToTask(taskDetails[1], "E");
+                break;
+            }
+            default:
+                break;
```

Comment:
I may be wrong, but perhaps it would be better to remove the default case if you are not doing anything useful inside. I understand the defensive logic with the `task != null` check but during the loading of the text file, if for some reason you were to read a `taskType` that was not `T`, `D` or `E`, would it be better to remove that line from the text file to avoid persisting invalid data?